### ISSUE TITLE
Can't login to the play game services - RESOLVED

### ISSUE THREAD

**Issue by GuillaumZ** [playgameservices/play-games-plugin-for-unity#1834]
This problem looks exactly like the one in these 2 threads : 
https://github.com/playgameservices/play-games-plugin-for-unity/issues/1826
and
https://github.com/playgameservices/play-games-plugin-for-unity/issues/1805
But the same solution given there doesn't fix my problem.

**Versions**
GIT Google Play Games for Unity version 0.9.39a
SDK Manager Google Play Games Service version 42 
I'm using Google app signing <IMPORTANT! Otherwise, you shouldn't have this problem>

**the problem**
When I'm building my game from Unity, directly into my android phone, I can login to Google Play Games.
Then I upload to Google Play store, in Closed Beta version and I download the game from there. And Now I can't connect to Google Play Games. The popup opens itself and then it fails silently.

**the solution that should work but doesn't work for me**
I looked for the SHA-1 numbers in the 3 places and they are **already** all  the same :
- In the keystore used with my game (SHA-1 checked with keytool.exe)
- in Google Play Developer Console  (the  SHA-1 from "Upload certificate")
- in the API Manager (https://console.developers.google.com - the Only SHA-1 there is, next to my game ID)

Anybody else is living this nightmare ?

**Comment by idurvesh** [comment]
I am having exactly same issue ....you geting this Debug msg as well? "DEBUG: Application is pausing, which disconnects the RTMP client. Leaving room."

**Comment by GuillaumZ** [comment]
Hi, how did you get this debug message ? 
In my case I don't think I've got any debug message readable by Unity. 
I've just got a boolean to "false" meaning that the authentication failed in this :

`Social.localUser.Authenticate ((bool success) => {
  if (success) {
  } else {
     // here I am
  }
});`

**Comment by idurvesh** [comment]
@GuillaumZ  Enable Debug play services in code  > make apk > install on device > open "Monitor" from "tools" folder in android sdk > conenct device to PC > run

**Comment by GuillaumZ** [comment]
Thank you. I didn't know that Monitor and it looks very usefull to debug.
But I don't have the same message than you.

I have this : 
"You have wrong OAUth2 related configurations, please check. Detailed error ; UNREGISTERED_ON_API_CONSOLE"
"OnSignInFailed()..."

So I checked again my SHA-1 on the API console and it's definitely right....
Still, I copied it from Google play console's App signing page's Upload certificate , SHA-1 certificate fingerprint and pasted it on the https://console.developers.google.com's App's credentials to be 200% sure it's the same one ... but it's still not working ...
Also, i find the link to my API console from Google Play console's Game services, my Game's first page where I can find the link and these infos that are telling me everything should be fine : 

API Console project
This game is linked to the API console project called 'XXXXX good pack's nameXXXX' ( + link to it)
The following APIs need to be turned on in the API console project for Games services to work:
<Checked in green>APIs required for basic Games Services to work
<Checked in green>APIs required for Saved Games to work

**Comment by idurvesh** [comment]
Infact I am getting error too, first time it runs properly next time when app auto sign in, it gives error like  [1145] BasicNetwork.performRequest: Unexpected response code 500 for https://www.googleapis.com/games/v1/players/me?language=en_IN
"Unable to load player xxxxx" 

YOur error seems to related to Console API, may be try agan creating services app and insert right certificate at the time of linking app.

**Comment by GuillaumZ** [comment]
![keystoreenygma](https://user-images.githubusercontent.com/8102439/28415848-cf4112d0-6d51-11e7-87e4-603466ebbc36.jpg)

So this is all the pieces of the puzzle.

It should work but I've got this error message : 
"You have wrong OAUth2 related configurations, please check. Detailed error UNREGISTERED_ON_API_CONSOLE"

**Comment by GuillaumZ** [comment]
So I've started it all over : 
- I've deleted my API in the API Console 
- I've created a new Game in Google Play Console's Game service with my leaderboars and achievements
- I've linked this new Game service to my app's package name
- I've uninstalled and re-installed the latest Google Play game Plugin for Unity
- I've pasted the Resources of the game from Google Play Game's Game Service.
- Then I went back on the API console to check the SAH-1 and Oh surprise, the name was, this time, different !
- I've tried to change it with the "good" on from my Keystore and Google Play Console's app signing's page BUT .... But !!! 
- ... But it doesn't want to take it telling me something like this : 

> This SHA1 is already used by an  ID client OAuth2 Android in this project or another one

 This "other one" project is now deleted ... so ... It looks like the nightmare doesn't want to end :-(

Perhaps it was going to work now ... I have now to contact google to see if they can give me another SHA1 ....

**Comment by idurvesh** [comment]
@GuillaumZ  Go to IAM in API Console (left side three horizontal lines), search for old project and in settings press "Shut Down"

**Comment by GuillaumZ** [comment]
Thank you, it's what I did. 
But It appears  that deleting a project in the API manager takes 30 days to "really" be deleted (as a popup says so, when you delete it)

As I didn't want to wait : 
- I created a new Google play project. With a new package name and a new keystore, in order **to get a new SHA1** for my game.
- I linked the app to my game in the Game service  in Google's play console
- I went into the API manager to put this new SHA1 in the new ID Client OAuth 2.0 ... Again, it was not the good one there, so I pasted the new one instead (the one from my keystore which is the same than the one in the Google play's app signing's page called "Upload certificate SHA1"). Like everyone is telling here this is the thing to do, I was quite optimistic that it would fix the problem now.

And then, I tried the game again by downloading it from the store (as always, everything is working fine when I'm building it directly from Unity to the phone) ... and ... Oh .... no ... **it's still not working !** ... and still the message "

> You have wrong OAuth2 related configurations, please check. Detailed error : UNREGISTERED_ON_API_CONSOLE

", when it tries to connect to Google Play Games ... When everything as been checked multiple times...

I also tried in "Open" Beta tests instead of "Closed", but it's not working either.

**Comment by idurvesh** [comment]
That is sad, I am stuck at similar kind issue...Have you eanbled Beta testing in Services package?

**Comment by GuillaumZ** [comment]
@idurvesh  I just tried, but it's still not working.
Also my Services package's "Game" is published so, apparently, it's not mandatory to activate it.
It's useful when your Play game service's game is in a draft status.

Good luck to you... maybe it's related. I'm gonna take a break. I'll try again new stuffs tomorrow, for now I can't think of any other things to try :-/

**Comment by GuillaumZ** [comment]
WOW
[RESOLVED]
I did the contrary of what I've been reading here (If I understood right!) : 
I copied/pasted the "App signing certificate" "SHA-1 certificate fingerprint" - Instead of the "Upload certificate" "SHA-1 certificate fingerprint" which is the one from my keystore, inside the API ID client OAuth 2.0 ... and now it's working !!!!

![keystoresolution](https://user-images.githubusercontent.com/8102439/28427236-d56445c8-6d75-11e7-9e1d-93da5f19d832.jpg)

**Comment by GuillaumZ** [comment]
Just to add a thing : so now, google play games is no more connecting in local. But it's connecting successfuly when it's been imported into the google play store (Google App signing does exactly what it's supposed to do : it changes your SHA1 from the "upload certificate" one, to the "App signing" one)

**Comment by ki4chan** [comment]
I am having the same problem. When I linked my app It had App signing SHA1 in API manager by default. Then the app was trying to sign in but failing every time. I replaced Upload Certificate with App Signing Certificate, now even signing in screen doesn't pop up.

One more question, are you guys using API key feature from API manager(the option above OAuth 2.0 client ID)?

**Comment by idurvesh** [comment]
Connect device to desktop > run device montor from android sdk's tools >
check what logs it throws

**Comment by ki4chan** [comment]
okay. I have never used log cat before so I am confused with all the logs. How to find which error is related to my game?

**Comment by idurvesh** [comment]
Search with "google" "your package name" or "GP4"

**Comment by green4you** [comment]
in my case,
i don't use app signing.

how to solve it?

**Comment by idurvesh** [comment]
Check if your mail id is added as beta tester...Check if you have init
services from unity, check if your OAth 2 has App signing certificate
rightly entered

**Comment by ki4chan** [comment]
The issue has been resolved and everything is working fine now. After trying 4-5 times I got to know why me and a lot of people having this problem, especially after Google's new App Signing feature. I believe most of the people are following old methods or old tutorials to implement google sign-in feature. Let me tell you what I was doing wrong all the time. After implementing Sign-in, achievements, leaderboards, I build the APK and transferred it to my android device and checked if it is working or not. And it wasn't working. I replaced upload certificate with App signing certificate then sign-in was working but the score was not updating on leaderboard it was getting closed automatically. 

So, in short, after building APK don't transfer it to your device. Upload it as beta testing then download it. Everything will work perfectly.

**Comment by LucaMazzoncini** [comment]
Oh yes work perfectly. Tnx
But how can I test it?Do I need to upload the app, every time i want to make a test?

**Comment by ki4chan** [comment]
Yes upload it in beta testing then only you'll be able to log in and Achievement/leaderboard will work perfectly.

**Comment by jsaraiva** [comment]
Just curious (as I haven't had the opportunity to test this yet): did you try to add another Linked app, this one using the SHA-1 fingerprint of your upload key?
In practice, you would end up having two such Linked apps: one with the fingerprint of the App Signing certificate, and the other one with the fingerprint of your Upload Key certificate (name this one "My App debug" or something, so you can easily remove it later).

As I said, didn't test this, but maybe it'll work (and you wouldn't have to be exchanging certificates back and forth).

**Comment by ki4chan** [comment]
I didn't get what you trying to say but I published it 4-5 times(after implementing everything again with new SHA1 etc) and they are still in my dashboard. But later I realized that publishing is not necessary to test it in beta or alfa. If things are not working in beta and Alfa you can delete them they will be gone from the dashboard.

**Comment by LucaMazzoncini** [comment]
Ok, I'll try to be more clear.
I'm developing an application in Unity with Google Play Services apis. I was trying to test the login phase but I got only failures (false return on Social.localUser.Authenticate((bool success)). So, searching for solutions, I stumbled on your solution, and it worked well.
My question is, do I need to load the testing apk on google dashboard every time I want to test google play services functionalities? Because if it is the case it's gonna be a nightmare. The application I designed is basically all founded on Google login, so almost everything it's gonna do is dependant on that phase and, apparently, to be tested it will need the pubblication on Google Dashboard.
Please tell me there's a easier way to test applications!

**Comment by jsaraiva** [comment]
Hi again.

I'll try to explain what I meant.

What I was saying was: you are probably getting an error in Authenticate because that app that you are using (locally) is signed with your Upload key [UK]. On the other hand, you probably configured your app in the Google Play Developer Console to have a Linked app (in the G.P. Dev. Console, see Game Services -> your game -> Linked apps) with a SHA1 signature that matches the key [GK] used by Google App Signing (see GPDevConsole -> All Applications -> your game -> App signing). So, Google's API services are expecting requests from an app with a signature [GK], but is receiving requests from an app with a signature [UK] (because you're testing locally, and so your app hasn't gone through the motions of Google removing your Upload key [UK] and replacing it with [GK]).

What I was asking was if you had tried to add _another_ Linked app, with the same details (same package name, etc), _but_ with the SHA-1 signature that corresponds to your Upload key (name this Linked app "<your game> debug", or something). In practice, you'd end up with 2 Linked apps: one with the Google key [GK] (you'll leave this one there forever), and another one with your Upload key [UK] (you can remove this one after you're sure you won't be testing your app locally anymore).
This way, you should be able to test your login both locally (with your app signed with [UK]) and after publishing (with the app signed by [GK]). I think.

Right now, the same is happening to me. I haven't been able to test this yet, because I have to change my Upload key and I'm still waiting for a response from Google. But from what I've researched, this should work fine.

Hope this helps!

Cheers!

**Comment by reflectivelayer** [comment]
Yes it does worked when you use the Upload key.
I have 3 apps linked.It the same app with keys for Google, upload and debug

**Comment by sunderam1996** [comment]
Today I uploaded my game.

I have checked that these keys are same =
- In the keystore used with my game (SHA-1 obtained from keytool.exe)
- In Google Play Developer Console (the SHA-1 obtained from "Upload certificate")
- In the API Manager (https://console.developers.google.com - Credentials)

And, "Sign-In" is not working properly. What can I do to solve this Issue.

**Comment by sunderam1996** [comment]
Now everything is solved 👍

**Comment by Opeth001** [comment]
Please how did you salve it ?

**Comment by sunderam1996** [comment]
Hello, @Opeth001 
I just followed the above steps and added missing code to the Android XML File. And, published the Google Play Games Services of the game from the Google Developer Console.
Refer #1970

**Comment by ki4chan** [comment]
@sunderam1996  
How are you testing it? By downloading it from play store or by directly Installing the build APK file? Download It from play store then check if the sign is working or not.

**Comment by Opeth001** [comment]
Hello, @sunderam1996 
Thank you for the quick reply :)

but i still  getting connection failed :( 
which  XML did you modified and w what did you added exactly ? 
I Need you help guys im getting this problem from 2 weeks now

**Comment by Opeth001** [comment]
ill Give all My setails so it will be better to understand my problem :(
i've added  GooglePlayGamesPlugin-0.9.41.unitypackage to my app unity Android.
i dont have any problem while wuilding the APK but i cant Sign in .
Social.localUser.Authenticate((bool success) =>  {  if (success) { }  else { // HERE I AM }   });  

I added 2 associated applications ( SHA1 Upload certificate &  SHA1 App signing certificate ) , 
I added my Tester Gmail in Google Play Console , i even tried to upload my app in Google Play and installing it again but same problem ( Connection failed )

here is my Unity Google Play Configuration ( https://img4.hostingpics.net/pics/960202Capture.png )

My Google Play Inizialization Script :  `void Start()
    {
        PlayGamesClientConfiguration config = new PlayGamesClientConfiguration.Builder()
          .RequestEmail()
          .RequestServerAuthCode(false)
          .AddOauthScope("email")
          .RequestIdToken()
          .Build();
 
        PlayGamesPlatform.InitializeInstance(config);
        PlayGamesPlatform.DebugLogEnabled = true;
        PlayGamesPlatform.Activate();
   }
 
    public void SignIn()
    {
        Social.localUser.Authenticate((bool success) =>
        {
            if (success)
            {
             Debug.Log("Connected");
 
            }
            else
             Debug.Log("Not Connected"); // here i am
        });
    }`

**Comment by MrColon6** [comment]
@Opeth001  I have the same issue and i am unable to sign in, please tell me if your thing is resolved!

**Comment by Opeth001** [comment]
Hi @MrColon6 ,
i Salved the Problem by deleting all my plugins ( in Plugins Folder ) and installing GPGs again 9.42 .
you need to create  3 Linked Applications in Google Play Consol ( Dont Create them from Google API Consol ), they will automatically added to the Google API Consol  ,
1) Android linked app using your SHA1 Upload ,
2) Android linked app using your SHA1 Signed ,
3) Web linked app using your Google Play Link  , ( https://play.google.com/store/apps/details?id=YOUR PKG NAME ) 

Go to Google API Consol 
![capture](https://user-images.githubusercontent.com/32723106/32815381-1b1f168e-c9b4-11e7-9893-99ecf1b3b65d.PNG)

And Finaly You Need to Add  your SHA1 here.
![capture2](https://user-images.githubusercontent.com/32723106/32815410-44e4faf6-c9b4-11e7-800e-5ac21c596b5a.PNG)

and everything will work Perfectly.

Good Luck!

**Comment by edesdan** [comment]
I'm having exactly the same problem, I have tried to add a new credential in the Google API Console Tool with the SHA1 from the Google Sign Certificate without success! This is very annoying. Anybody have a proper solution? I don't want to re-create app, change package name eccc like suggested in some of the previous comments. This is serious issue, I cannot publish my game, damn me I have opted-in for this google signing nightmare..

**Comment by ColonL** [comment]
@edesdan Yup I have the same doubt. we have no idea which application are we saving on cloud and also it'll be hard to manage data like leaderboards and achievements. 

Please Help us Guys!

**Comment by edesdan** [comment]
I managed to solve the problem by removed the linked app and recreating it again in the google console. I think for me the problem was that i created the linked app before joining the app google signing security program. Now it is ok even if I can only test leaderboards and achievements once the game is released as alpha, beta or prod.

**Comment by LucaMazzoncini** [comment]
. 2cl. Io l

Inviato da smartphone Samsung Galaxy.
-------- Messagidgio originale --------Da: LalithPrasad S <notifications@github.com> Data: 20/11xdv7 /17  5:52 PM  (GMT+01:00) A: playgameservices/play-games-plugin-for-unity <play-games-plugin-for-unity@noreply.github.com> Cc: LucaMazzoncini <mazzonciniluca@gmail.com>, Comment <comment@noreply.github.com> Oggetto: Re: [playgameservices/play-games-plugin-for-unity] Can't login to the play game services - RESOLVED (#1834) 
@edesdan Yup I have the same doubt. we have no idea which application are we saving on cloud and also it'll be hard to manage data like leaderboards and achievements.
Please Help us Guys!

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/playgameservices/play-games-plugin-for-unity","title":"playgameservices/play-games-plugin-for-unity","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/playgameservices/play-games-plugin-for-unity"}},"updates":{"snippets":[{"icon":"PERSON","message":"@ColonL in #1834: @edesdan Yup I have the same doubt. we have no idea which application are we saving on cloud and also it'll be hard to manage data like leaderboards and achievements. \r\n\r\nPlease Help us Guys! "}],"action":{"name":"View Issue","url":"https://github.com/playgameservices/play-games-plugin-for-unity/issues/1834#issuecomment-345756626"}}}

**Comment by Opeth001** [comment]
hi @edesdan ,
i posted the method that i used , and everything is working for me (SignIn, Achievements , leaderboards ..... )
if you have the failed login message ( dont forget that u cant use the gpgs into the unity emulator u have to login from ur device ) you have to create 2 linked  android apps one with sha1 upload and another with sha1 signed certificate.

( everything is working for me )

**Comment by ColonL** [comment]
Yeah Problem Solved, your app should be on play store to access play services. every thing works fine once you release it!

**Comment by bitbionic** [comment]
Thank you for this work. This post saved me hours of headache!.. @edesdan I had the same issue as you in that I created the app before joining google app signing. I also had to enable the Google Drive API. I'm curious - with the new google signing API are you guys able to test Play Services locally - or do you upload an alpha build and then pull it down with the new keys and remote debug?

**Comment by edesdan** [comment]
No chance to test it locally, you need to upload it as alpha/beta. I just
got a confirmation from Google. I think that's a great limitation. I will
not join the program for my next app.

**Comment by bitbionic** [comment]
@edesdan I just figured out how to test locally.. You just need to register your SHA 1 for your Upload key with Google APIs Credentials. So you have an OAuth cert for the App Key and for the Upload Key. Now I'm able to test play services locally.

**Comment by Opeth001** [comment]
@edesdan 
Yes you can test it localy ! ( i doing it by build and run from unity to my device  )
just dont forget to configure your  Upload  SHA1 Key .

**Comment by PN1019** [comment]
@GuillaumZ @idurvesh what do you mean by "Enable debug play services in code"?Is this some checkbox or u mean to do debugging in code?

**Comment by vaspotapov** [comment]
@PN1019  
> @GuillaumZ @idurvesh what do you mean by "Enable debug play services in code"?Is this some checkbox or u mean to do debugging in code?

you can just do this

PlayGamesClientConfiguration config = new PlayGamesClientConfiguration.Builder()
			.Build();
		PlayGamesPlatform.InitializeInstance(config);
		PlayGamesPlatform.DebugLogEnabled = true;
		PlayGamesPlatform.Activate();

**Comment by dkrazor** [comment]
> WOW
> [RESOLVED]
> I did the contrary of what I've been reading here (If I understood right!) :
> I copied/pasted the "App signing certificate" "SHA-1 certificate fingerprint" - Instead of the "Upload certificate" "SHA-1 certificate fingerprint" which is the one from my keystore, inside the API ID client OAuth 2.0 ... and now it's working !!!!
> 
> ![keystoresolution](https://user-images.githubusercontent.com/8102439/28427236-d56445c8-6d75-11e7-9e1d-93da5f19d832.jpg)
> 
> I just lost last night and today on this ^^'

**This configuration mode is working in:**
Unity 2019.2.0f1
GooglePlayGamesPlugin-0.9.64.unitypackage

best!!!

**Comment by real-mikhail** [comment]
Thanks a lot to everyone here! You've just saved me!

**Comment by dkrazor** [comment]
Good

Em qui., 27 de fev. de 2020 às 08:52, Mehmet Ozdemir <
notifications@github.com> escreveu:

**Comment by QuocHieuNguyen** [comment]
Anyone here ? I still encounter this problem, both SHA1 (upload & app signing) is not working.

**Comment by itsMohammedThaier** [comment]
Hello guys
I have a problem with
 _Sign-in_


**this is the details:**
.
1_ I add play games services package to my unity project

2_ I add this script:

```
 using UnityEngine;
 using GooglePlayGames;
 using GooglePlayGames.BasicApi;
 using UnityEngine.SocialPlatforms;
 
  public class mainMenuEvents : MonoBehaviour
 {
     public void Start()
     {
         ///this to configuration//////
         PlayGamesClientConfiguration config = new PlayGamesClientConfiguration.Builder().Build();
         PlayGamesPlatform.InitializeInstance(config);
         PlayGamesPlatform.DebugLogEnabled = true;
         PlayGamesPlatform.Activate();
 
 
 
         ///this for silent Sign-In
         PlayGamesPlatform.Instance.Authenticate(success => 
         {
             if (success)
             {
                 Debug.Log("Loged in success");
                 successObject.enabled = true;  ///this object has Image telling that you logged in successfully 
             }
 
             if (!success)
             {
                 Debug.Log("Failed");
                 failedObject.enabled = true;  ///this object has Image telling the logged in failed
             }
         });
 
     }
 }
```

3_ I create application in Google play console > all applications &  Adding my game in App Release< Internal test track & putting my accounts as testers

4_I create application in **Game services** and linked it with my game and authorized it to **API Console**

5_In the **API Console** i put the **SHA1** like the **SHA1** I get from **Google Play Console < My application < App signing Upload SHA1 key**

6_ I make 1 **leadboard** to get **my application resource** and put it in **unity Google Play Games Set Up**

### What i did next?
I build and run the game on my phone wich is have the allowed accounts to test the game but the game google play services and Sign-In did't work :( 
When i run the game in unity i can't sign in so i get "Sign in failed" and show up image saying "Failed" So this means my code is right but when i run it in my phone its don't give me Failed or success (The sign in method don't even work)

### The most logical reasons:
1_The google play services don't allow me to test (But i add my accounts to the testers account so i'm not sure about this)
2_The configuration maybe don't work
3_The Assign key have something wrong
4_ In every tutorial i see they download **Google play services** in the SDK folder by **Android Studio < Sdk Manager** but i have some problems makes me can't download it to **unity SDK Folder** & the reason is : **_When i download it its save in folder named "Extras" and when unity build the android application it use only 4 folder ("Platform","Platform-tools","Build-tools","tools")_**

I had done exactly like [this google tutorial](https://codelabs.developers.google.com/codelabs/playservices_unity/index.html?index=..%2F..index) but i got the same problem
& did exactly like  those tutorials [1](https://www.youtube.com/watch?v=0LGs1Xtt_1I&t=1488s) and [2](https://www.youtube.com/watch?v=M6nwu00-VR4&t=612s) but didn't help

### Pleas any one can help 
### If this didn't work i will lose 2 month of work :<

**Comment by itsMohammedThaier** [comment]
> Anyone here ? I still encounter this problem, both SHA1 (upload & app signing) is not working.

Did you resolve it?

**Comment by itsMohammedThaier** [comment]
### Resolved
The problem was a bug in **Play Games Services 10.9** so i download **Play Games Services 10.10** and its work!

**Comment by kimseitner** [comment]
> inside the API ID client OAuth 2.0

oh wow you saved my day - thank you!

**Comment by djmuhlestein** [comment]
I'm having this issue with 0.10.12

I'm on 0.10.12.  Getting the dreaded APP NOT CORRECTLY CONFIGURED TO USE GOOGLE PLAY GAME SERVICES.
DEVELOPER_ERROR

* The sha1 matches
* the app package matches.
* I've tried on the play store side to set the credentials to published instead of relying on the tester accounts ( which have the same error)
* I've tried using my upload certificate and I also tried using the play store signing certificate and uploading my app and then downloading it to test.

Beating my head over this.  It all looks like it should work.  Any suggestions on how to further debug to find out really why the config is being denied?  Could it be the way I'm trying to use or initialize the plugin? (It's pretty simple and similar or equal to what everyone else is doing as far as I can see..)


I need the authCode because I want to use it to authenticate with firebase. 

       PlayGamesClientConfiguration config = new PlayGamesClientConfiguration.Builder()
            .RequestIdToken()
            .RequestServerAuthCode(false)
            .Build();

        PlayGamesPlatform.InitializeInstance(config);
        PlayGamesPlatform.DebugLogEnabled = true;
        PlayGamesPlatform.Activate();   

If I don't try to retrieve the server auth code, the call to Social.localUser.Authenticate(...) returns success instead and the player is authenticated with Google Play.   (But then I can't use it to sign in to Firebase.. unless I'm missing something and there is another way.)

**Comment by h-sigma** [comment]
> 
> 
> I'm having this issue with 0.10.12
> 
> I'm on 0.10.12. Getting the dreaded APP NOT CORRECTLY CONFIGURED TO USE GOOGLE PLAY GAME SERVICES.
> DEVELOPER_ERROR
> 
>     * The sha1 matches
> 
>     * the app package matches.
> 
>     * I've tried on the play store side to set the credentials to published instead of relying on the tester accounts ( which have the same error)
> 
>     * I've tried using my upload certificate and I also tried using the play store signing certificate and uploading my app and then downloading it to test.
> 
> 
> Beating my head over this. It all looks like it should work. Any suggestions on how to further debug to find out really why the config is being denied? Could it be the way I'm trying to use or initialize the plugin? (It's pretty simple and similar or equal to what everyone else is doing as far as I can see..)
> 
> I need the authCode because I want to use it to authenticate with firebase.
> 
> ```
>    PlayGamesClientConfiguration config = new PlayGamesClientConfiguration.Builder()
>         .RequestIdToken()
>         .RequestServerAuthCode(false)
>         .Build();
> 
>     PlayGamesPlatform.InitializeInstance(config);
>     PlayGamesPlatform.DebugLogEnabled = true;
>     PlayGamesPlatform.Activate();   
> ```
> 
> If I don't try to retrieve the server auth code, the call to Social.localUser.Authenticate(...) returns success instead and the player is authenticated with Google Play. (But then I can't use it to sign in to Firebase.. unless I'm missing something and there is another way.)

SAME ISSUE
I have tried practically everything.

**Comment by creepteks** [comment]
> 
> 
> I'm having this issue with 0.10.12
> 
> I'm on 0.10.12. Getting the dreaded APP NOT CORRECTLY CONFIGURED TO USE GOOGLE PLAY GAME SERVICES.
> DEVELOPER_ERROR
> 
>     * The sha1 matches
> 
>     * the app package matches.
> 
>     * I've tried on the play store side to set the credentials to published instead of relying on the tester accounts ( which have the same error)
> 
>     * I've tried using my upload certificate and I also tried using the play store signing certificate and uploading my app and then downloading it to test.
> 
> 
> Beating my head over this. It all looks like it should work. Any suggestions on how to further debug to find out really why the config is being denied? Could it be the way I'm trying to use or initialize the plugin? (It's pretty simple and similar or equal to what everyone else is doing as far as I can see..)
> 
> I need the authCode because I want to use it to authenticate with firebase.
> 
> ```
>    PlayGamesClientConfiguration config = new PlayGamesClientConfiguration.Builder()
>         .RequestIdToken()
>         .RequestServerAuthCode(false)
>         .Build();
> 
>     PlayGamesPlatform.InitializeInstance(config);
>     PlayGamesPlatform.DebugLogEnabled = true;
>     PlayGamesPlatform.Activate();   
> ```
> 
> If I don't try to retrieve the server auth code, the call to Social.localUser.Authenticate(...) returns success instead and the player is authenticated with Google Play. (But then I can't use it to sign in to Firebase.. unless I'm missing something and there is another way.)

I'm having the same issue.

**Comment by JairoGLoz** [comment]
I have "properly" configured everything and still have the following issue:
- When I log in to google play using my gmail account which is also my developer account, everything works fine: login, achievements and leaderboard. But...
- When I use any other of my tester gmails, login fails silently... I have 7 different tester accounts and none of them is able to access google play services.

**Comment by Oldfagg** [comment]
Hey. I want to upload Unity GPS but those files dont wanna be checked. Can someone help me out?
![photo_2021-07-01_00-30-56](https://user-images.githubusercontent.com/65858024/124034255-a56be200-da03-11eb-8d07-5ec4f038d9ca.jpg)

**Comment by shokanshi** [comment]
> I have "properly" configured everything and still have the following issue:
> 
> * When I log in to google play using my gmail account which is also my developer account, everything works fine: login, achievements and leaderboard. But...
> * When I use any other of my tester gmails, login fails silently... I have 7 different tester accounts and none of them is able to access google play services.

Don't know if it is applicable to you but have you tried turning "Enable anti-piracy" off under 

Play Game Services -> Configuration -> Credentials

I was having issues getting leaderboards to show up on test accounts till I have that turned off.

**Comment by rameezfayyaz** [comment]
Hi,
I had the same issue and after wasting at least 3-4 days and trying all above solutions, the thing that worked for me is this:
**Apart from the list of emails in Internal or any other release type, the email list of testers for Play Games Services is seperate, for whatever reason. Its under Play Games Services>Setup and Management>Testers in the play console. Once I added the tester emails in there as well, the login and leaderboards and achievements worked like a charm.**
My other settings are as follows if it helps anyone:
1. In both play console and cloud console I use the SHA-1 from App Signing Key certificate instead of Upload Key certificate
2. I use the Webclient ID in Android Setup inside Unity
3. I use an additional Android client with SHA-1 from the Upload Key certificate but the option "Use For New Installs" is unchecked for this and checked for App Signing Certificate
4. I keep Enable anti-piracy option unchecked while game is in testing as this can result in unwanted authorization rejections.

**Comment by FadeSoft** [comment]
Same problem in 2023 help!!!
